lua/toggleterm/ui.lua

```diff
@@ -10,10 +10,16 @@ local api = vim.api
 local persistent = {}
 local origin_window
 
-function M.save_window_size()
-  -- Save the size of the split before it is hidden
-  persistent.width = vim.fn.winwidth(0)
-  persistent.height = vim.fn.winheight(0)
+--- Save the size of a split window before it is hidden
+--- @param window number
+--- @param direction string
+function M.save_window_size(direction, window)
+  window = window or nvim_get_current_win()
```

Comment:
this looks like its missing a `vim.api.nvim_get_current_win`, so will error if the window passed in is nil